brian2/codegen/optimisation.py

```diff
@@ -38,12 +38,12 @@ def evaluate_expr(expr, ns):
 
     Example
     -------
-    >>> assumptions = {'sin': DEFAULT_FUNCTIONS['sin'].pyfunc,
-    ...                'pi': DEFAULT_CONSTANTS['pi'].value}
+    >>> assumptions = {'exp': DEFAULT_FUNCTIONS['exp'].pyfunc,
+    ...                'inf': DEFAULT_CONSTANTS['inf'].value}
```

Comment:
Any reason for change from sin to exp etc.? Not that it matters much, just curious.

Reply:
I briefly explain this in the commit message: our docttest relied on the fact that `np.sin(np.pi/2) == 1.0`, which is reasonable but not guaranteed (results are only guaranteed to be < 4 epsilons away from the correct value). With the previous versions it matched 1.0 exactly, but with the newer, faster implementations in `numpy` 1.25 this is no longer the case (even with older versions, `np.sin(0.0)` or `np.sin(np.pi)` did not return 0.0). So I replaced it with `exp(-inf)` which I think is guaranteed to be exactly zero – at least it works with both numpy 1.24 and numpy 1.25 :)